pandas/core/tools/datetimes.py

```diff
@@ -400,6 +403,12 @@ def _convert_listlike_datetimes(
             arg = arg.tz_convert(None).tz_localize("utc")
         return arg
 
+    elif isinstance(arg_dtype, ArrowDtype) and arg_dtype.kind == "M":
```

Comment:
`pd.ArrowDtype(pa.date32())` also has kind == "M".  i think for this we need `pa.types.is_timestamp(arg_dtype.pyarrow_dtype)`.  or maybe `arg_dtype.type is Timestamp`